tests/masking_test.py

```diff
@@ -12,51 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
 from functools import partial
 import itertools as it
 from unittest import SkipTest
 
 import numpy as np
 from absl.testing import absltest, parameterized
-from jax.interpreters.masking import shape_as_value, ShapeError, \
-  parse_spec, Poly, Mon
-from jax import numpy as jnp, test_util as jtu, mask, vmap, jit, grad, lax, \
-  shapecheck, api
+
+from jax.interpreters.masking import (shape_as_value, parse_spec, ShapeError,
+  Poly, Mon, eval_polymorphic_shape, remap_ids, UniqueIds)
+from jax import (numpy as jnp, test_util as jtu, mask, vmap, jit, grad, lax,
+                 core as jc, shapecheck, safe_map, safe_zip, unzip2,
+                 tree_flatten, tree_unflatten, tree_map)
 from jax.config import config
+from jax.lax.lax import _identity
 from jax.numpy.lax_numpy import _polymorphic_slice_indices
+from jax.random import uniform, PRNGKey
 from jax.scipy.special import expit
+from operator import add, sub
+import scipy.stats
 
 config.parse_flags_with_absl()
 
+map = safe_map
+zip = safe_zip
 
-# These are 'manual' tests for masking. The more exhaustive,
-# more systematic tests should live in lax_test.py.
+
+# TODO:
+# These should be only the 'manual' tests for masking.
+# Move the more exhaustive, systematic tests into lax_test.py.
 
 def constant_poly(c):
   return Poly({Mon(): c})
 
-class ShapesTest(jtu.JaxTestCase):
+class MaskingTest(jtu.JaxTestCase):
 
   @parameterized.parameters([
-      ['(m, n)', 'ShapeSpec(m, n)'],
-      ['(m * n)', 'ShapeSpec(m n)'],
-      ['m * n', 'ShapeSpec(m n)'],
-      ['(m * n,)', 'ShapeSpec(m n)'],
-      ['(3, m)', 'ShapeSpec(3, m)'],
-      ['(10, m)', 'ShapeSpec(10, m)'],
-      ['(-10, m)', 'ShapeSpec(-10, m)'],
-      ['(3 * m)', 'ShapeSpec(3 m)'],
-      ['m', 'ShapeSpec(m)'],
-      ['', 'ShapeSpec()'],
-      ['m + n', 'ShapeSpec(m + n)'],
-      ['m + n * k', 'ShapeSpec(m + k n)'],
-      ['m + 3 * k', 'ShapeSpec(3 k + m)'],
-      ['', 'ShapeSpec()'],
-      ['_', 'ShapeSpec(_)'],
+    ['(m, n)', 'ShapeSpec(m, n)'],
```

Comment:
This is a minor nit, but can you revert the indentation on these lines so that they appear the same in the diff?

Reply:
Resolved